### DISCUSSION TITLE
SSG with fallback: true, regenerate only one page

### DISCUSSION THREAD

**Discussion by miguel9554** [vercel/next.js#16541] [Help]
'm working with a mostly static site, with a small group of pages pre-rendered and the majority rendered on the first request, by using `fallback: true`.

Of this group of static pages, about 1 page per day changes (this is really an upper bound, sometimes it's 1 page per week). To change just that one page, I need to rebuild the entire site, because it was generated on the first request, and it won't be built again unless I regenerate the site. This is bad, since I need to regenerate 90% of the site just because of 1 page.

Is there an alternaitve to this? Is it possible to erase the page once it has been generated, so in the next request it's generated again?

Using [incremental static generation](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) doesen't seem like the optimal solution, because every page would need to be rebuilt after the revalidation timeout, it's not exactly what I'm looking for.

**Comment by alanqchen** [comment]
Until Next.js adds re-generating by listening to events in the future, I think the current incremental static generation will have to be a necessary compromise and was the route I personally took.

**Reply by michrome** [reply]
Cross linking this to #16488 which is a request for the same: regeneration based on an event, to cut short the wait for a revalidation period.
